pkg/controller/nodelifecycle/node_lifecycle_controller.go

```diff
@@ -155,8 +161,9 @@ type Controller struct {
 	computeZoneStateFunc       func(nodeConditions []*v1.NodeCondition) (int, ZoneState)
 
 	knownNodeSet map[string]*v1.Node
-	// per Node map storing last observed Status together with a local time when it was observed.
-	nodeStatusMap map[string]nodeStatusData
+	// Per Node map storing last observed healthy data, including Status and Lease,
+	// together with a local time when it was observed.
+	nodeHealthMap map[string]nodeHealthData
```

Comment:
Since you're touching it anyway - this should probably be map[string]*nodeHealthData.